sansio_lsp_client/client.py

```diff
@@ -108,19 +220,23 @@ def _send_response(
     ) -> None:
         self._send_buf += _make_response(id=id, result=result, error=error)
 
+    # response from server
     def _handle_response(self, response: Response) -> Event:
         assert response.id is not None
         request = self._unanswered_requests.pop(response.id)
 
-        # FIXME: The errors have meanings.
         if response.error is not None:
-            raise RuntimeError("Response error!\n\n" + pprint.pformat(response.error))
+            err = ResponseError.parse_obj(response.error)
+            err.message_id = response.id
+            return err
```

Comment:
Any reason why not `raise err`?

Reply:
`ResponseError` - not an exception, but an error event from the server.

Reply:
Are there any cases when it would be useful to handle a `ResponseError` differently from an exception?

Reply:
This is a response to a request. User needs to receive a `message_id` along with the error-string.